### PR TITLE
Add type definitions for recommended + x-teaser

### PR DESCRIPTION
**Pull request by adgad**
Adds `RecommendedReference` (which has a reference to a content ID), and a `Recommended` which uses x-teaser props for a resolved recommended link.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +57,13 @@ interface ImageSource {
 }
 ```
 
-TODO: clarify that i mean Strong cannot have an ancestor of Strong etc
+### Teaser
+
+```ts
+import { TeaserProps as Teaser } from '@financial-times/x-teaser'
```
Comment (adgad) [inline README.md:63]: Does this make sense as an approach? The alternative is copy/paste all the type definitions from x-teaser (~20 lines or so)

File: README.md [hunk 2]
```diff
@@ -272,17 +279,31 @@ interface Pullquote extends Parent {
...
+
+### `Recommended`
+```ts
+interface Recommended extends Node {
+	type: "recommended"
+	heading?: string
+	teaserTitleOverride?: string
+	teaser: Teaser
+}
+```
```
Comment (chee) [inline README.md:306]: great

File: README.md (hunk 1 above)
Comment (adgad) [reply README.md]: Ooh yeah good point about versioning! That makes sense.

I'm only copying the types we need/use for recommended link teasers for now. I think at some point _someone_ should do a major version of x-teaser with a better data model (things like their image definition, and metaLinks are a bit weird). It requires thinking about all the non-article use cases though, so lets not couple ourselves to that.

Comment (chee) [reply README.md]: > Ooh yeah good point about versioning! That makes sense.

forever i think i'll be profoundly affected by the trouble faced by jake for polyfill.io in the javascript spec's definition of whitespace as "the latest unicode spec's definition of whitespace"

> lets not couple ourselves to that.

i'm sure we'll get there eventually ❤️

File: README.md [hunk 3]
```diff
@@ -56,7 +57,64 @@ interface ImageSource {
 }
 ```
 
-TODO: clarify that i mean Strong cannot have an ancestor of Strong etc
```
Comment (chee) [inline README.md:59]: how did this get all the way down there !!!

File: README.md [hunk 4]
```diff
@@ -56,7 +57,64 @@ interface ImageSource {
...
+type TeaserContentType =
+  | 'article'
+  | 'video'
+  | 'podcast'
+  | 'audio'
+  | 'package'
+  | 'liveblog'
+  | 'promoted-content'
+  | 'paid-post'
+
```
Comment (chee) [inline README.md:72]: is this used anywhere else? we can inline this on line 103 instead

**Review by chee — APPROVED** [review]
eat

**Pull request merged**